buildbot/master/files/config/factories.py

```diff
@@ -103,6 +104,95 @@ def make_step(self, command):
         return step_class(**step_kwargs)
 
 
+class StepsYAMLParsingStep(buildstep.ShellMixin, buildstep.BuildStep):
+    """
+    Step which resolves the in-tree YAML and dynamically add test steps.
+    """
+
+    haltOnFailure = True
+    flunkOnFailure = True
+
+    def __init__(self, builder_name, environment, yaml_path):
+        self.builder_name = builder_name
+        self.environment = environment
+        self.yaml_path = yaml_path
+
+    @defer.inlineCallbacks
+    def run(self):
+        try:
+            full_yaml_path = os.path.join(self.build.workdir, self.yaml_path)
+            print_yaml_cmd = "cat {}".format(full_yaml_path)
```

Comment:
I believe the command will be run in the workdir by default, so just passing `self.yaml_path` as a relative path should work. Also, I don't think the current code wouldn't work for Windows, as the `os.path.join` would take place on the master (and use `/`), but the generated command would actually execute on the worker.